cherrybomb-engine/src/lib.rs

```diff
@@ -23,44 +23,51 @@ fn verbose_print(config: &Config, required: Option<Verbosity>, message: &str) {
 
 pub async fn run(config: &Config) -> anyhow::Result<Value> {
     verbose_print(config, None, "Starting Cherrybomb...");
-
-    // Reading OAS file to string
     verbose_print(config, None, "Opening OAS file...");
-    let oas_file = match std::fs::read_to_string(&config.file) {
-        Ok(file) => file,
-        Err(e) => {
-            return Err(anyhow::anyhow!("Error reading OAS file: {}", e));
-        }
-    };
 
-    // Parsing OAS file to JSON
-    verbose_print(config, None, "Parsing OAS file...");
-    let oas_json: Value = match serde_json::from_str(&oas_file) {
-        Ok(json) => json,
-        Err(e) => {
-            return Err(anyhow::anyhow!("Error parsing OAS file: {}", e));
-        }
-    };
-
-    // Parsing JSON to OAS struct
-    verbose_print(config, Some(Verbosity::Debug), "Creating OAS struct...");
-    let oas: OAS3_1 = match serde_json::from_value(oas_json.clone()) {
-        Ok(oas) => oas,
-        Err(e) => {
-            return Err(anyhow::anyhow!("Error creating OAS struct: {}", e));
+    match config.file.extension() {
+        Some(ext) => {
+            verbose_print(config, None, "Reading OAS file...");
+            let oas_file = match std::fs::read_to_string(&config.file) {
+                Ok(file) => file,
+                Err(e) => return Err(anyhow::Error::msg(format!("Error opening OAS file: {}", e))),
+            };
+            let oas_json: Value = match ext.to_str() {
+                Some("json") => {
+                    verbose_print(config, None, "Parsing OAS file...");
```

Comment:
"Parsing OAS json file..." to make better use of this stdout line.
likewise other cli output in this function.

Reply:
What do you suggest?